yellowbrick/text/dispersion.py

```diff
@@ -95,34 +120,91 @@ def fit(self, text):
         text : list
             Should be provided as a list of documents that contain
             a list of words in the order they appear in the document.
+
+        y : ndarray or Series of length n
+            An optional array or series of target or class values for
+            instances. If this is specified, then the points will be colored
+            according to their class.
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
         """
 
+        if y is not None:
+            self.classes_ = np.unique(y)
+        elif y is None and self.labels is not None:
+            self.classes_ = np.array([self.labels[0]])
+        else:
+            self.classes_ = np.array([self.NULL_CLASS])
+
         # Create an index (e.g. the y position) for the target words
         self.target_words_ = np.flip(self.words, axis=0)
         if self.ignore_case:
             self.target_words_ = np.array([w.lower() for w in self.target_words_])
 
         # Stack is used to create a 2D array from the generator
-        points = np.stack(self._compute_dispersion(text))
-        self.draw(points)
+        self.points_target = np.stack(self._compute_dispersion(text, y))
+
+        self.points = np.stack(zip(self.points_target[:,0].astype(int),
```

Comment:
It looks like `self.points` could just be `points`, since it isn't used outside of `fit()`.